### ISSUE TITLE
Video distorted

### ISSUE THREAD

**Issue by MorDeCaii** [wtsnz/obs-ios-camera-source#3]
Tryed to add video source from iPhone 5s and iPad on Windows 10, but in both cases (with any resolution) video is distorted and jumpy. Same result on laptop.

**Comment by MorDeCaii** [comment]
@wtsnz Here is screenshot from mobile app and GIF of what is on the PC screen (no matter USB 2.0 or 3.0):
![img_0991](https://user-images.githubusercontent.com/11280671/38772609-ef5fb56a-4043-11e8-8a70-35b24d1de5d1.PNG)
![0e1ec5b1bdcb21c73d3014e4ac30d89e](https://user-images.githubusercontent.com/11280671/38772618-167e5926-4044-11e8-8da7-d5945cdf9f4d.gif)

**Comment by wtsnz** [comment]
Oh wow! That is something!

Can I ask what hardware you have on your computer? (Graphics card etc)

**Comment by MorDeCaii** [comment]
@wtsnz, of course: 
On PC: CPU: i7-4770K, GPU: AMD Radeon HD 7970, RAM: 16GB
On Laptop: CPU: i5-4210U, GPU: Nvidia Geforce GT 730M, RAM: 8GB

**Comment by wtsnz** [comment]
Thanks for getting back to me. Odd I was wondering it there was issues with a particular GPU manufacturer so it's good to know that it's nothing specific to AMD or Nvidia!

I don't really know what the cause is just yet, but I'm working on adding hardware decoder support that may fix it. I'm hoping to get a new version of the plugin ready in a few hours, and it would be awesome if you could try it out 😄

**Comment by wtsnz** [comment]
I've committed some changes that look for a different decoder, there is an installer available here https://ci.appveyor.com/project/wtsnz/obs-ios-camera-source/build/1.0.30/artifacts

If you could give it a go, and check out the OBS Log file (In OBS Studio, Help -> Log Files -> View Current Log), and it'll tell you whats going on, it would be awesome to see the contents of that too!

**Comment by MorDeCaii** [comment]
For now, it seems like hardware decoder not work:

12:40:28.137: [obs-ios-camera-plugin] Started listening for devices
12:40:28.147: [obs-ios-camera-plugin] Activating
10:35:29.943: [obs-ios-camera-plugin] Started listening for devices
10:35:29.945: User added source 'iOS Camera' (ios-camera-source) to scene 'Game'
10:35:29.951: [obs-ios-camera-plugin] Activating
10:35:39.285: [obs-ios-camera-plugin] Found hardware accelerator: h264_dxva2
10:35:39.286: [obs-ios-camera-plugin] No hardware decoder for h264_dxva2
10:35:39.286: [obs-ios-camera-plugin] Couldn't find hardware decoder. Using software decoder
10:35:39.286: [obs-ios-camera-plugin] Using decoder: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 (h264)

**Comment by ecjo** [comment]
Hi!

I have the same issue, although Im using my MacBook Pro, and connect it through USB to my iPhone, with OBS Studio. Same issue with the image though. Any ideas of a solution?
Thanks

**Comment by wtsnz** [comment]
Thanks for the report @ecjo.

It's an interesting one, as I've still not got to the bottom of it... I've just moved countries last week so I'm slowly getting back into the swing of things 😄

There's no USB hubs involved at all?

**Comment by wtsnz** [comment]
I have managed to replicate this on my iPad 3 (the first retina iPad) and it appears to only happen when the video resolution on the iPad is set at 1080x1920@30 and higher. If I set the video resolution less that that (720x1280@30) it works as expected.

Sorry for the delay of getting into this issue, moving countries is fun 😅  I'm now looking into the cause of this issue.

**Comment by wtsnz** [comment]
Thanks for the comment vfmradio. So it doesn't work on your iPhone 5? 

I'm currently sourcing some older iOS devices to test this issue, as I'm starting to think that there is a hardware difference in some of the older iOS devices that is causing this. 
My theory is that devices older than the iPhone 6, and the iPad 3 experience this issue. 
If you are experiencing it, then try lowering the resolution - this works on my iPad 3.

**Comment by vfmradio** [comment]
I tried to change the resolution, but it does not change anything. I wanted to give a second life to the old iphone, but I have to use my new device (I will be happy with any solutions if they appear!

**Comment by Dikyx** [comment]
Hi,
I bought the app a couple hours ago. I found this issue report on Github after experiencing the same problem.

**iPhone :** 
Model : iPhone 5S
iOS version : 12.1.1

**Computer :** 
CPU : I7 6700k
GPU : GTX 1080Ti
OS : Windows 10 Version 10.0.17134 Build 17134
OBS version : 21.1.0
OBS ios camera source plugin version : v2.3.1 (windows)
iTunes version : 12.9.1.4

Workarounds : 
Lowering resolution didn't work for me.

Nothing new here but I posted it anyway, just in case :p
